go/README.md

```diff
@@ -4,27 +4,70 @@ This builder runs the `go` tool (`go build`, `go test`, etc.)
 after placing source in `/workspace` into the `GOPATH` before
 running the tool.
 
-This functionality is not necessary if you're building using
+# Using `Golang` and [Go Modules](https://github.com/golang/go/wiki/Modules)
+
+This Builder (`gcr.io/cloud-builders/go`) is not necessary if you're building using
 [Go modules](https://github.com/golang/go/wiki/Modules), available
-in Go 1.11+, and you can **build with the standard
-[`golang`](https://hub.docker.com/_/golang) image on Dockerhub
-instead:**
+in Go 1.11+. You can **build** with the `golang` image (not a Builder) from [Dockerhub](hub.docker.com/library/golang) and Google's Container Registry [mirror](mirror.gcr.io/library/golang):
```

Comment:
We configure GCB to use the GCR mirror by default, so there's no need to explicitly request the mirrored version. Just referencing `golang` will pull through the GCR mirror if available, and not if not.

Reply:
Interesting. How does this square with: https://issuetracker.google.com/issues/137170318

Reply:
That I don't know, I'd have to talk to the GCR team to understand more.

In general I think we should not ask users to request from `mirror.gcr.io` directly, but instead just configure their Docker installation to use the mirror, or in the case of GCB, just rely on it already configuring that.